File: 2.cpp
#include <iostream>
#include <random>
#include <vector>
#include <algorithm>
#include <map>

using namespace std;

int main() {
    setlocale(LC_ALL, "Russian");
    // Задание 1: Инициализация массива из 10 случайных чисел в диапазоне [0, 100]
    const int n = 10;
    vector<double> arr(n);
    random_device rd;
    minstd_rand gen(rd());
    uniform_real_distribution<> dis(0, 100);
    for (int i = 0; i < n; ++i) {
        arr[i] = dis(gen);
    }
    cout << "Задание 1: Массив из 10 случайных чисел в диапазоне [0, 100]:" << endl;
    for (const auto& num : arr) {
        cout << num << " ";
    }
    cout << endl;

    // Задание 2: Вычисление среднего значения и суммы квадратов разностей
    double sum = 0;
    for (int i = 0; i < n; ++i) {
        sum += arr[i];
    }
    double average = sum / n;
    double sum_of_squares = 0;
    for (int i = 0; i < n; ++i) {
        sum_of_squares += (arr[i] - average) * (arr[i] - average);
    }
    cout << "Задание 2: Сумма квадратов разностей: " << sum_of_squares << endl;

    // Задание 3: Замена первого и последнего разрядов чётных элементов с чётным номером
    for (int i = 0; i < n; i += 2) {
        int num = static_cast<int>(arr[i]);
        if (num >= 10 && num < 100) {
            int first_digit = num / 10;
            int last_digit = num % 10;
            arr[i] = last_digit * 10 + first_digit;
        }
    }
    cout << "Задание 3: Массив после замены первого и последнего разрядов:" << endl;
    for (const auto& num : arr) {
        cout << num << " ";
    }
    cout << endl;

    // Задание 4: Сдвиг элементов массива на одну позицию вправо
    rotate(arr.begin(), arr.begin() + 1, arr.end());
    cout << "Задание 4: Массив после сдвига на одну позицию вправо:" << endl;
    for (const auto& num : arr) {
        cout << num << " ";
    }
    cout << endl;

    // Задание 5: Инициализация массива из 50 случайных чисел в диапазоне [-10, 10]
    const int m = 50;
    vector<int> arr2(m);
    uniform_int_distribution<> dis2(-10, 10);
    for (int i = 0; i < m; ++i) {
        arr2[i] = dis2(gen);
    }
    cout << "Задание 5: Массив из 50 случайных чисел в диапазоне [-10, 10]:" << endl;
    for (const auto& num : arr2) {
        cout << num << " ";
    }
    cout << endl;

    // Подсчёт количества вхождений каждого числа в массиве
    map<int, int> count;
    for (int i = 0; i < m; ++i) {
        count[arr2[i]]++;
    }
    cout << "Задание 5: Количество вхождений каждого числа в массиве:" << endl;
    for (const auto& pair : count) {
        cout << pair.first << ": " << pair.second << endl;
    }

    // Удаление повторяющихся элементов из массива
    sort(arr2.begin(), arr2.end());
    arr2.erase(unique(arr2.begin(), arr2.end()), arr2.end());
    cout << "Задание 5: Массив после удаления повторяющихся элементов:" << endl;
    for (const auto& num : arr2) {
        cout << num << " ";
    }
    cout << endl;

    return 0;
}

File: 2.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        // Task 1: Initialization of an array with 10 random numbers in the range [0, 100]
        int n = 10;
        double[] arr = new double[n];
        Random rand = new Random();
        for (int i = 0; i < n; ++i) {
            arr[i] = rand.nextDouble() * 100;
        }
        System.out.println("Task 1: Array of 10 random numbers in the range [0, 100]:");
        for (double num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();

        // Task 2: Calculation of the average value and the sum of the squares of the differences
        double sum = 0;
        for (int i = 0; i < n; ++i) {
            sum += arr[i];
        }
        double average = sum / n;
        double sumOfSquares = 0;
        for (int i = 0; i < n; ++i) {
            sumOfSquares += Math.pow(arr[i] - average, 2);
        }
        System.out.println("Task 2: Sum of the squares of the differences: " + sumOfSquares);

        // Task 3: Replacement of the first and last digits of even elements with an even index
        for (int i = 0; i < n; i += 2) {
            int num = (int) arr[i];
            if (num >= 10 && num < 100) {
                int firstDigit = num / 10;
                int lastDigit = num % 10;
                arr[i] = lastDigit * 10 + firstDigit;
            }
        }
        System.out.println("Task 3: Array after replacing the first and last digits:");
        for (double num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();

        // Task 4: Shift of array elements to the right by one position
        double temp = arr[n - 1];
        for (int i = n - 1; i > 0; --i) {
            arr[i] = arr[i - 1];
        }
        arr[0] = temp;
        System.out.println("Task 4: Array after shifting to the right by one position:");
        for (double num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();

        // Task 5: Initialization of an array with 50 random numbers in the range [-10, 10]
        int m = 50;
        int[] arr2 = new int[m];
        for (int i = 0; i < m; ++i) {
            arr2[i] = rand.nextInt(21) - 10;
        }
        System.out.println("Task 5: Array of 50 random numbers in the range [-10, 10]:");
        for (int num : arr2) {
            System.out.print(num + " ");
        }
        System.out.println();

        // Counting the occurrences of each number in the array
        Map<Integer, Integer> count = new HashMap<>();
        for (int i = 0; i < m; ++i) {
            count.put(arr2[i], count.getOrDefault(arr2[i], 0) + 1);
        }
        System.out.println("Task 5: Count of occurrences of each number in the array:");
        for (Map.Entry<Integer, Integer> entry : count.entrySet()) {
            System.out.println(entry.getKey() + ": " + entry.getValue());
        }

        // Removing duplicate elements from the array
        Arrays.sort(arr2);
        int j = 0;
        for (int i = 0; i < m - 1; ++i) {
            if (arr2[i] != arr2[i + 1]) {
                arr2[j++] = arr2[i];
            }
        }
        arr2[j++] = arr2[m - 1];
        arr2 = Arrays.copyOf(arr2, j);
        System.out.println("Task 5: Array after removing duplicate elements:");
        for (int num : arr2) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
}